htdocs/core/actions_massactions.inc.php

```diff
@@ -47,6 +47,13 @@
 }
 $error = 0;
 
+// Note: list of strings for objectclass could be extended to accepted/expected classes
+'
+@phan-var-force "Societe"|"Contact"|"ExpenseReport"|"Partnership"|"Holiday"|"ConferenceOrBoothAttendee"|"Facture"|"CommandeFournisseur" $objectclass
```

Comment:
$objectclass can be any Dolibarr business object here.
Is it possible to have instead a generic object class, may be CommonObject class ?

Reply:
I find often that putting CommonObject alone generates issues with the typing as some fields are missing and expected by the algorithm.

I propose to "add" CommonObject and also leave the other options that are tested for in the code.  In case there is a typing issue it will indicate which classes are not compatible.

Ideally - in some future - the code would do a test using "instanceof" rather than relying on the 'element' field and the typing hints.
That implies risks of course: the "instanceof" condition could be too strict vs. what the code was actually expecting and introduce some bugs in the process.

Reply:
> I propose to "add" CommonObject and also leave the other options that are tested for in the code. In case there is a typing issue it will indicate which classes are not compatible.

You mean having this
@phan-var-force CommonObject|"Societe"|"Contact"|"ExpenseReport"|"Partnership"|"Holiday"|"ConferenceOrBoothAttendee"|"Facture"|"CommandeFournisseur" $objectclass   ?

Reply:
Yes.

FYI, phan has more than 50 notices for this file.

For instance, we have:
```
htdocs\core\actions_massactions.inc.php:1288 PhanUndeclaredProperty Reference to undeclared property \CommonObject->price
htdocs\core\actions_massactions.inc.php:1289 PhanUndeclaredProperty Reference to undeclared property \CommonObject->price_min
```

There is no check on the object type: it is implied by the 'updateprice' action.

Reply:
That is because we have the following in the configuration for phan:

```
dev/tools/phan/config.php:              'obj' => '\CommonObject',     // Deprecated
```` 

We could check what the result is if this is commented in a separate PR (I guess there will be some notifications that pop up).